java/runtime/src/main/java/org/ray/runtime/DefaultRayRuntimeFactory.java

```diff
@@ -25,9 +27,17 @@ public RayRuntime createRayRuntime() {
       if (rayConfig.runMode == RunMode.SINGLE_PROCESS) {
         runtime = new RayDevRuntime(rayConfig, functionManager);
       } else {
+        RunManager manager = null;
+        if (rayConfig.getRedisAddress() == null) {
+          manager = new RunManager(rayConfig);
+          manager.startRayProcesses(true);
+        }
+        RayNativeRuntime.setup(rayConfig);
+
         if (rayConfig.workerMode == WorkerType.DRIVER) {
-          runtime = new RayNativeRuntime(rayConfig, functionManager);
+          runtime = new RayNativeRuntime(manager, rayConfig, functionManager);
         } else {
+          Preconditions.checkState(manager == null);
           runtime = new RayMultiWorkerNativeRuntime(rayConfig, functionManager);
```

Comment:
I think it's the time to add `SingleWorkerNativeRuntime` to avoid use `RayNativeRuntime` for driver.